### PR TITLE
Remove lpsolve external libraries

### LINKED ISSUE #141: Remove lpsolve external libraries
**Is your feature request related to a problem? Please describe.**
External libraries consume space in the repository and have to be updated manually.

**Describe the solution you'd like**
Remove `lpsolve` from `external` folder and download them in `cmake` before building the package or whenever needed. 

See how this was done for eigen https://github.com/GeomScale/volume_approximation/pull/131

### PR DESCRIPTION
**Pull request by ManavShah05**
Created a cmake file to automatically download and build lp solver. The source code was found from sourceforge.net. Closes #141

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+cmake_minimum_required(VERSION 3.13)  # CMake version check
```
Comment (vissarion) [inline CMakeLists.txt]: why do we need such a file in the root of the project ?

**Review by vissarion — COMMENTED** [review]
Thanks for this PR!

It is a good start in general. There are some issues though. 

First you have to branch from the current develop, now it seems that there are some commits here not related to this PR. A merge with current develop should fix that issue.

Second, and most important, you have to remove lp_solve library from external and see if your approach is working. Actually this is the main reason of doing this.

**Comment by ManavShah05** [comment]
I had a quick question. Currently, the cmake file is downloading and building the lp_solve in the test folder. After removing the files of lp_solver from the external folder, do I need to relocate the build to the external folder or does it work if it stays in the test folder?

**Comment by ManavShah05** [comment]
Also, I have been receiving this error a lot and I cant seem to find a solution online. I wanted to ask if you could help me with this.

Whenever I try to make the file in the test folder, the file gets downloaded and extracted but the system is not able to change the name of the directory.

Permission Denied

**Review by pradeexsu — COMMENTED** [review]
fine

**Comment by ManavShah05** [comment]
I have removed the lp_solve files from the external folder and made the cmake file such that it download the source code in the test folder called _deps. I have modified the CMakeLists.txt file in the test folder to change the path of lp_solver to the _deps folder. I tested the file and it built successfully and passed all the tests. 

Also, I wanted to add that the issue that I described here: https://github.com/GeomScale/volume_approximation/pull/149#issuecomment-810025988 only happens when the program is run in WSL.

**Comment by vissarion** [comment]
> I have removed the lp_solve files from the external folder and made the cmake file such that it download the source code in the test folder called _deps. I have modified the CMakeLists.txt file in the test folder to change the path of lp_solver to the _deps folder. I tested the file and it built successfully and passed all the tests.

It seems that the tests are failing on cicleci see https://app.circleci.com/pipelines/github/GeomScale/volume_approximation/604/workflows/735a37af-5a38-4129-859a-71accc4db5ef/jobs/815

> Also, I wanted to add that the issue that I described here: [#149 (comment)](https://github.com/GeomScale/volume_approximation/pull/149#issuecomment-810025988) only happens when the program is run in WSL.

**Comment by ManavShah05** [comment]
I am not sure what is causing the error in these files as they are getting build correctly in my IDE, however, they are failing on the server tests. 
Screenshot from 2021-03-31 16-36-29
Screenshot from 2021-03-31 16-37-01

**Comment by vissarion** [comment]
You maybe want to try to check the CI environments locally e.g. https://circleci.com/docs/2.0/local-cli/ and https://github.com/nektos/act

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -61,7 +61,7 @@ include("Boost.cmake")
 GetBoost()
 
 include("lp_solver.cmake")
-Get_LP_solver()
+Get_LP_Solver()
 
 find_library(LP_SOLVE NAMES liblpsolve55.so liblpsolve55.dylib PATHS /usr/lib/lp_solve /opt/local/lib)
```
Comment (warmachine86) [inline test/CMakeLists.txt:66]: You have not removed this line which should be removed. Maybe that's why they are failiing on the the server tests and getting built correctly on your machine.

**Comment by ManavShah05** [comment]
@vissarion I have passed the circleci build

**Comment by ManavShah05** [comment]
I studied all the failed tests and found that while constructing the cran_package from cran_gen we are copying all the necessary files. Due to this, genCRANpkg.R is trying to copy the lp solver from its older location in the external folder. As we have removed the lp solver from the external folder we need to extract it from the build folder which we make after executing the CMakeLists.txt file in the test folder. Since in the automated test we are simply copy pasting the files, is it possible to add cmake commands to build lpSolver and then copy the files?

**Comment by vissarion** [comment]
This is fixed by https://github.com/GeomScale/volesti/pull/183

**Pull request closed without merging**